### PR TITLE
Implement select_process_and_model

### PR DESCRIPTION
**Pull request by pcanal**
This replace https://github.com/celeritas-project/celeritas/pull/165 which github closed automatically (due to some branch maintenance on my side) and I can not re-open.

### CODE REVIEW

**Comment by sethrj** [comment]
When a PR is undergoing review, please refrain from force pushing as it makes it harder to track changes, in addition to confusing github (which apparently is pretty easy since both you and @mrguilima both did it in the same day 😂 )

**Comment by mrguilima** [comment]
So the rebase to master should be done only at the very end, once all comments have been addressed?

**Comment by sethrj** [comment]
@mrguilima Since we're doing squash instead of merge commits it's ok to merge master into your branch rather than rebasing, if you think the updates on master affect your merge requests.

File: src/physics/base/PhysicsStepUtils.i.hh
Comment (sethrj) [inline src/physics/base/PhysicsStepUtils.i.hh:174]: Can you move this lengthy comment to the function documentation above line 130 (and leave a one-line comment here referencing the function docs)? A large block of text in the middle of the function makes it harder to read.

**Comment by sethrj** [comment]
One last comment @pcanal . Thanks!

File: test/physics/base/PhysicsTestBase.hh
Comment (sethrj) [inline test/physics/base/PhysicsTestBase.hh:98]: @pcanal Sorry I just noticed this addition -- the `PhysicsTestBase` class is also used by device tests (which of course don't use the host RNG). Can you move this up to the PhysicsStepUtils harness, which is host-only?

**Comment by sethrj** [comment]
#160 also touched the physics step utils. I can help resolve conflicts if needed.

**Comment by sethrj** [comment]
@pcanal The other reason why force-pushing is annoying during review is that clicking on notifications on github brings you to an error page:

Screen Shot 2021-03-07 at 20 54 57
Screen Shot 2021-03-07 at 20 55 11

**Comment by pcanal** [comment]
>  The other reason why force-pushing is annoying during review is that clicking on notifications on github brings you to an error page

:( indeed that is annoying :(

**Comment by pcanal** [comment]
(on the other hand, not being able to force push also means having to merge with master rather than rebase and I find it a bit harder to do conflict resolution correctly/reliably with a merge (where you are more or presented with a final diff of the 2) rather than a rebase (where your presented with just a single commit incremental change compare to the master).

Anyway, I hold off on force push :)

**Comment by sethrj** [comment]
> (on the other hand, not being able to force push also means having to merge with master rather than rebase and I find it a bit harder to do conflict resolution correctly/reliably with a merge (where you are more or presented with a final diff of the 2) rather than a rebase (where your presented with just a single commit incremental change compare to the master).
Interesting, I usually find the reverse -- because when I rebase, sometimes each commit will result in a conflict at the same line with the upstream. Merging always means you have a single conflict resolution step, and the history of the PR reflects the actual development (i.e. if it worked on one of the commits before the merge, it still works -- the same can't be said if you're rebasing, since "changes don't step on each other" doesn't necessarily mean "builds and passes after fixing conflicts").

Anyway, I've been on both sides of this issue so I understand where you're coming from. In my mind, "one PR = one commit in the historical record" helps alleviate the anxiety of having extra merges in a branch -- in early git days before GitHub/GitLab "merge master into branch and push" was the common way and it left history in a shambles.

I appreciate your understanding!

**Review by sethrj — APPROVED** [review]
Looks good, thanks @pcanal !

**Pull request merged**
